crates/bin/pd/src/migrate/testnet74.rs

```diff
@@ -0,0 +1,159 @@
+//! Contains functions related to the migration script of Testnet72
+
+use anyhow;
+use cnidarium::{EscapedByteSlice, Snapshot, StateDelta, StateRead, StateWrite, Storage};
+use futures::StreamExt as _;
+use jmt::RootHash;
+use penumbra_app::app::StateReadExt as _;
+use penumbra_app::SUBSTORE_PREFIXES;
+use penumbra_dex::SwapExecution;
+use penumbra_num::Amount;
+use penumbra_proto::penumbra::core::component as pb;
+use penumbra_proto::StateWriteProto;
+use penumbra_sct::component::clock::{EpochManager, EpochRead};
+use prost::Message;
+use std::path::PathBuf;
+
+use crate::testnet::generate::TestnetConfig;
+
+/// Updates arb execution output amounts to include the input amount instead
+/// of reporting only profit (see #3790).
+async fn fix_arb_execution_outputs(delta: &mut StateDelta<Snapshot>) -> anyhow::Result<()> {
+    let mut stream = delta.prefix_raw("dex/arb_execution/");
```

Comment:
Could offload deserialization to `prefix_proto` would straighten up the rest of the migration